src/modelconfig.hpp

```diff
@@ -303,7 +310,27 @@ class ModelConfig {
          * @param basePath 
          */
     void setBasePath(const std::string& basePath) {
-        this->basePath = basePath;
+        if (!FileSystem::isLocalFilesystem(basePath)) {
+            // Cloud filesystem
+            this->basePath = basePath;
+        } else if (basePath.at(0) == '/') {
+            // Full path case
+            this->basePath = basePath;
+        } else {
+            // Relative path case
+            if (this->rootDirectoryPath.empty())
+                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Using model relative path without setting configuration directory path.");
```

Comment:
Shouldn't we here throw an exception in case someone will change the order of operations setBasePath, setRootDirectory?

Reply:
Ok